salalib/alllinemap.cpp

```diff
@@ -173,55 +174,71 @@ std::tuple<std::unique_ptr<ShapeGraph>, std::unique_ptr<ShapeGraph>> AllLineMap:
    std::map<RadialKey,RadialSegment> radialsegs;
 
    // now make radial segments from the radial lines... (note, start at 1)
-   for (i = 1; i < m_radial_lines.size(); i++) {
-      if (m_radial_lines[i].vertex == m_radial_lines[i-1].vertex) {
-         if (m_radial_lines[i].ang == m_radial_lines[i-1].ang) {
+   auto iter = m_radial_lines.begin();
+   auto prevIter = m_radial_lines.begin();
+   ++iter;
+   for (;iter != m_radial_lines.end();) {
+      if (iter->vertex == prevIter->vertex) {
```

Comment:
That whole block line 181 to 191 could be rewritten as
```suggestion
      if (iter->vertex == prevIter->vertex && iter->ang != prevIter->ang) {
                radialsegs.insert(std::make_pair( (RadialKey)(*iter), (RadialSegment)(*prevIter)));
      }          
```